src/Rule/CompareHandler.php

```diff
@@ -217,6 +220,8 @@ private function normalizeNumber(mixed $number): float
     {
         if ($number instanceof Stringable) {
             $number = (string) $number;
+        } elseif ($number instanceof DateTimeInterface) {
```

Comment:
DateTimeInterface should be compared to DateTimeInterface only. We should not even try to compare 42 with the date.